typescript-carousel/main.ts

```diff
@@ -0,0 +1,127 @@
+const images: string[] = [
+  './images/001.png',
+  './images/004.png',
+  './images/007.png',
+  './images/025.png',
+  './images/039.png',
+];
+
+let currNdx: number = 0;
+
+const $displayImage = document.querySelector(
+  '.display-image'
+) as HTMLImageElement;
+if (!$displayImage) throw new Error('.display-image query failed!');
+
+$displayImage.setAttribute('src', images[currNdx]);
+
+const $carouselContainer = document.querySelector(
+  '.carousel-container'
+) as HTMLDivElement;
+if (!$carouselContainer) throw new Error('.carousel-container query failed!');
+
+const $leftArrow = document.querySelector('.fa-chevron-left');
+if (!$leftArrow) throw new Error('.fa-chevron-left query failed!');
+const $rightArrow = document.querySelector('.fa-chevron-right');
+if (!$rightArrow) throw new Error('.fa-chevron-right query failed!');
+
+const $circles = document.querySelectorAll('.fa-circle');
+if (!$circles) throw new Error('.fa-circles query failed!');
+
+$carouselContainer.addEventListener('click', function (event: Event) {
+  if (event.target === $leftArrow) {
+    moveLeft();
+  }
+
+  if (event.target === $rightArrow) {
+    moveRight();
+  }
+
+  switch (event.target) {
+    case $circles[0]: {
+      clearInterval(intervalId);
+      intervalId = setInterval(intervalFunc, 3000);
+      clearCircles();
+      setNdxImageAndDot(0);
+      break;
+    }
+    case $circles[1]: {
+      clearInterval(intervalId);
+      intervalId = setInterval(intervalFunc, 3000);
+      clearCircles();
+      setNdxImageAndDot(1);
+      break;
+    }
+    case $circles[2]: {
+      clearInterval(intervalId);
+      intervalId = setInterval(intervalFunc, 3000);
+      clearCircles();
+      setNdxImageAndDot(2);
+      break;
+    }
+    case $circles[3]: {
+      clearInterval(intervalId);
+      intervalId = setInterval(intervalFunc, 3000);
+      clearCircles();
+      setNdxImageAndDot(3);
+      break;
+    }
+    case $circles[4]: {
+      clearInterval(intervalId);
+      intervalId = setInterval(intervalFunc, 3000);
+      clearCircles();
+      setNdxImageAndDot(4);
+      break;
+    }
+  }
+});
+
+let intervalId = setInterval(intervalFunc, 3000);
+
+function clearCircles(): void {
+  for (let i = 0; i < $circles.length; i++) {
+    $circles[i].classList.remove('fa-solid');
+    $circles[i].classList.add('fa-regular');
+  }
+}
```

Comment:
Now that I think about it, this function may be unnecessary. You have your `currNdx` globally available which tells you where you are. We should be able to reset the `fa-solid` or `fa-regular` `classList` based on this `currNdx` rather than looping through every circle every time we want to change one back to regular.